source/module_io/read_set_globalv.cpp

```diff
@@ -44,15 +44,54 @@ void ReadInput::set_globalv(Parameter& para)
                 para.sys.gamma_only_local = false;
             }
         }
+        /// set deepks_setorb
+        if (para.input.deepks_scf || para.input.deepks_out_labels)
+        {
+            para.sys.deepks_setorb = true;
+        }
+        switch (para.input.nspin)
+        {
+        case 4:
+            if (para.input.noncolin)
+            {
+                para.sys.domag = true;
+                para.sys.domag_z = false;
+            }
+            else
+            {
+                para.sys.domag = false;
+                para.sys.domag_z = true;
+            }
+            para.sys.npol = 2;
+            break;
+        case 2:
+        case 1:
+            para.sys.domag = false;
+            para.sys.domag_z = false;
+            para.sys.npol = 1;
+            para.input.lspinorb = false;
```

Comment:
move "para.input.* = *" to read_input_item_*.cpp

Reply:
the question has been finished.